test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+        {
+            Some(_resource_message) => _resource_message.as_str().map(|s| s.to_owned()),
+            None => Some(String::from("default")),
+        };
+        let _resource_id = match value
+            .get("resource")
+            .and_then(|resource| resource.get("id"))
+        {
+            Some(_resource_id) => _resource_id
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _resource_special_fields = SpecialFields::default();
+        let mut _resource = UResource::new();
+        _resource.name = _resource_name.to_owned();
+        _resource.instance = _resource_instance;
+        _resource.message = _resource_message;
+        if!(_resource_id == 0)
+        {
+        _resource.id = Some(_resource_id)
+    };
+
+        let ___resource = MessageField(Some(Box::new(_resource)));
+        let _special_fields = SpecialFields::default();
+    let mut _uuri:UUri = UUri::new();
+    if!(_authority_name.clone() == Some("default") && _authority_number_id.clone() == vec![0] && _authority_number_ip.clone() == vec![0]){
+        dbg!("authority is not default");
+        _uuri.authority = MessageField(Some(Box::new(_authority)));
+    }
+    _uuri.entity = ___entity;
+    _uuri.resource = ___resource;
+
+
+        Ok(WrapperUUri(_uuri))
+    }
+}
+#[derive(Default)]
+pub struct WrapperUAttribute(pub UAttributes);
+impl<'de> Deserialize<'de> for WrapperUAttribute {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+        // Conversion function from string to enum variant
+        fn from_str_comstatus(s: &str) -> UCode {
+            match s {
+                "ABORTED" => UCode::ABORTED,
+                "OK" => UCode::OK,
+                "CANCELLED" => UCode::CANCELLED,
+                "UNKOWN" => UCode::UNKNOWN,
+                "INVALID_ARGUMENT" => UCode::INVALID_ARGUMENT,
+                "DEADLINE_EXCEEDED" => UCode::DEADLINE_EXCEEDED,
+                "NOT_FOUND" => UCode::NOT_FOUND,
+                "ALREADY_EXISTS" => UCode::ALREADY_EXISTS,
+                "PERMISSION_DENIED" => UCode::PERMISSION_DENIED,
+                "UNAUTHENTICATED" => UCode::UNAUTHENTICATED,
+                "RESOURCE_EXHAUSTED" => UCode::RESOURCE_EXHAUSTED,
+                "FAILED_PRECONDITION" => UCode::FAILED_PRECONDITION,
+                "OUT_OF_RANGE" => UCode::OUT_OF_RANGE,
+                "UNIMPLEMENTED" => UCode::UNIMPLEMENTED,
+                "INTERNAL" => UCode::INTERNAL,
+                "UNAVAILABLE" => UCode::UNAVAILABLE,
+                "DATA_LOSS" => UCode::DATA_LOSS,
+                &_ => todo!(),
+            }
+        }
+        fn from_str_priority(s: &str) -> UPriority {
```

Comment:
I think the [same suggestion](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1566222380) applies here.